### PR TITLE
Make loading synchronous.

### PR DESCRIPTION
**Pull request by L3MON4D3**
Rip out all `vim.schedule` and luv-callbacks from loaders.
The problem with asynchronous loading is that race-conditions may occur when snippets are added after a call to `load()` (see #281)

@uga-rosa @leiserfg Could you take a look at these changes? I'm not completely sure if it's okay to remove the `vim.schedule`, but we don't call any `vim.api` in there so I guess it is..?

### CODE REVIEW

**Review by leiserfg — APPROVED** [review]

**Comment by L3MON4D3** [comment]
Nice, okay.
Loading the entirety of `vim-snippets` _does_ introduce a noticeable delay on startup, so we should probably notify users to switch to `lazy_load` or the method described in the wiki before this is merged.

**Comment by uga-rosa** [comment]
Is this really asynchronous?

**Comment by L3MON4D3** [comment]
It's possible I mixed the two up, I looked at a [definition](https://www.merriam-webster.com/dictionary/synchronous) and assumed that it's the same in CS, but it seems like you're right.
I'll change the title, ty👍

**Comment by uga-rosa** [comment]
As @L3MON4D3 said above, even if this change slows down the startup time, we can use lazy load, so I don't think it will be a problem.

**Comment by L3MON4D3** [comment]
Another idea to speed up loading:
Don't actually parse the snippets while loading, just store the string-representation into the snippet-table. The snippet will be parsed when it is triggered.

**Comment by uga-rosa** [comment]
That's an interesting idea. I'd like to try it with larger snippets and see if the latency is a problem. (Considering the speed of Lua, I don't think it will be a problem)

**Comment by L3MON4D3** [comment]
You can get a pretty close approximation by binding `ls.lsp_expand(snippet_body)` to some key.
In my experience it's instantaneous.

**Comment by uga-rosa** [comment]
I tested it a bit and it seems to be fine.
Do you want to implement this feature in this PR?

**Comment by L3MON4D3** [comment]
Yup, although I won't have time for a few weeks

**Comment by L3MON4D3** [comment]
On master, `load()`ing `honza/vim-snippets` takes 1ms of startuptime (everything loaded asynchronally), here the startup is delayed by the full 55ms (3ms with `lazy_load()`).

What does work, however, is wrapping `load()` in `vim.schedule`. That allows one to both have <1ms delay during startup and do some work on the loaded snippets.
```lua
vim.schedule(function()
	require("luasnip.loaders.from_snipmate").load()
end)
```

I'm going to post as much in `Breaking Changes`, that way users/maintainers of neovim-configs already know what's going on/can prepare for when this PR is merged.

**Comment by leiserfg** [comment]
@L3MON4D3  try 
```vim
vim.cmd [[au BufRead * ++once require("luasnip.loaders.from_snipmate").load() ]]
```

**Comment by L3MON4D3** [comment]
Right, or like that 👍
(Does the autocommand have any additional benefit with `load`? I'll change the recommendation to it, if it has)

**Comment by leiserfg** [comment]
It should take the same amount of time but should be less noticeable for the user (and won't appear in the startup time for those that are in that kind of competition).

**Comment by L3MON4D3** [comment]
Ah sorry, I meant benefit compared to wrapping `load` in `vim.schedule`.

**Pull request merged**
